core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java

```diff
@@ -134,4 +170,27 @@ public void setEmpty(boolean empty) {
     }
     return aclEntryList;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof DefaultAccessControlList)) {
+      return false;
+    }
+    DefaultAccessControlList that = (DefaultAccessControlList) o;
+    return mOwningUser.equals(that.mOwningUser)
+        && mOwningGroup.equals(that.mOwningGroup)
+        && mUserActions.equals(that.mUserActions)
+        && mGroupActions.equals(that.mGroupActions)
+        && mMaskActions.equals(that.mMaskActions)
+        && mOtherActions.equals(that.mOtherActions);
```

Comment:
what about `mAccessAcl` and `mIsEmpty`?

Reply:
added

Reply:
actually `mAccessAcl` need not to be equal for the `DefaultAccessControlList` to be equal, it is just an external reference. so removed that from both `equals` and `hashCode`